lib/matplotlib/tests/test_backend_bases.py

```diff
@@ -50,30 +49,22 @@ def check(master_transform, paths, all_transforms,
     check(id, paths, tforms, offsets, facecolors[0:1], edgecolors)
 
 
-def test_get_default_filename():
-    try:
-        test_dir = tempfile.mkdtemp()
-        plt.rcParams['savefig.directory'] = test_dir
-        fig = plt.figure()
-        canvas = FigureCanvasBase(fig)
-        filename = canvas.get_default_filename()
-        assert filename == 'image.png'
-    finally:
-        shutil.rmtree(test_dir)
+def test_get_default_filename(tmpdir):
+    plt.rcParams['savefig.directory'] = tmpdir
+    fig = plt.figure()
+    canvas = FigureCanvasBase(fig)
+    filename = canvas.get_default_filename()
+    assert filename == 'image.png'
 
 
-def test_get_default_filename_already_exists():
+def test_get_default_filename_already_exists(tmpdir):
     # From #3068: Suggest non-existing default filename
-    try:
-        test_dir = tempfile.mkdtemp()
-        plt.rcParams['savefig.directory'] = test_dir
-        fig = plt.figure()
-        canvas = FigureCanvasBase(fig)
+    plt.rcParams['savefig.directory'] = tmpdir
+    fig = plt.figure()
+    canvas = FigureCanvasBase(fig)
 
-        # create 'image.png' in figure's save dir
-        open(os.path.join(test_dir, 'image.png'), 'w').close()
+    # create 'image.png' in figure's save dir
+    open(os.path.join(tmpdir, 'image.png'), 'w').close()
```

Comment:
`tmpdir.join('image.png').ensure()` (I think)

Reply:
not convinced it's worth using the pytest-specific api here.
fixed other points.